src/chapters/functions/why-create-functions.rst

```diff
@@ -0,0 +1,67 @@
+
+.. sourcecode:: js
+
+   function isPalindrome(str) {
+      
+      let reversed = '';
+      
+      for (let i = 0; i < str.length; i++) {
+         reversed = str[i] + reversed;
+      }
+
+      return reversed === str;
+   }
+
+While the variable name ``reversed`` is descriptive, giving us a sense of what is going on with the ``for`` loop, the function becomes even more readable when we break out the reversing behavior into a separate function.
+
+.. sourcecode:: js
+
+   function reverse(str) {
+      let reversed = '';
+      
+      for (let i = 0; i < str.length; i++) {
+         reversed = str[i] + reversed;
+      }
+      return reversed;
+   }
+
+   function isPalindrome(str) {
+      return reversed(str) === str;
+   }
+
+Aside from following the principle that functions should do only one thing, the logic within ``isPalindrome`` is more clear and self-descriptive. The function itself says, *a string is a palindrome if it is equal to its reverse*. To draw this same conclusion from the example above, without a ``reverse`` function, we are required to analyze more of the program's logic.
+
+Functions Reduce Complexity
+---------------------------
+
+Large programs can be broken down into smaller parts using functions. Imagine a car built out of a single, large piece of metal. Were such a car to break down, diagnosing the problem would be difficult, and fixing it nearly impossible. The mechanic would have to determine where the issue was, then cut apart the bad portion, create a custom-made replacement portion, and weld it into place. 
+
+The complexity of this situation becomes much less when a car is made up of lots of small parts, each of which can be tested and replaced individually. The same thing happens with code. While there are many other organizational units for programs---including modules, files, and packages---functions are the most basic and universal organizational tool. 
+
+Functions Enable Code Sharing
+-----------------------------
+
+.. todo:: add reference to modules chapter
+
+As you will soon learn, encapsulating behavior within a function makes it easy for you to share that behavior across files and even projects. This becomes a critically important feature when working on large programs that consist of dozens, or even hundreds of files of code, as all professional developers do regularly. 
+
+Functions Save Millions of Lives Every Day
+------------------------------------------
+
+Okay, not really. But the point is, functions are incredibly powerful tools that you will come to appreciate and find indispensable. Ask a professional programmer if they could do their job without functions, and the answer will be an emphatic "NO!" 
+
+While they are abstract and can be difficult to learn, we promise you that the work will be worth it. 
```

Comment:
Spin this.  Instead of affirming that functions are abstract or hard to learn, how about "While functions may seem abstract and difficult to learn at first, repeated practice will quickly lead to mastery.  We promise that your work will be worth it."